scripts/showcase/samplesheet-prep-test.py

```diff
@@ -403,17 +441,48 @@ def write_sample_sheets(sample_sheet_header_rows, sample_sheet_df, sample_sheet_
     sample_sheet_header_rows : list of lines used to create the sample sheet header
     sample_sheet_df : pd.DataFrame with sample information per row
     sample_sheet_dir : Path output directory to put SampleSheet_<Type>.csv
+    sample_type : A specific sample to filter by
+    index_len : A specific index length to filter by
+    index2_len : A specific index2 length to filter by
     """
     logger.info("Writing out sample sheets")
-    for sample_type, sample_sheet_type_df in sample_sheet_df.groupby('Type'):
+
+    # Filter sample sheet by sample
+    if sample_type is not None:
+        sample_sheet_df = sample_sheet_df.query("Type=='{}'".format(sample_type))
+    if sample_sheet_df.shape[0] == 0:
+        logger.error("After filtering for type '{}' we ended up with no rows in the sample sheet".format(sample_type))
+        sys.exit(1)
+
+    # Filter sample sheet by index length
+    if index_len is not None:
+        sample_sheet_df = sample_sheet_df.query("index_len=={}".format(index_len))
+    if sample_sheet_df.shape[0] == 0:
+        logger.error("After filtering for index of len '{}' we ended up with no rows in the sample sheet".format(index_len))
+        sys.exit(1)
+
+    # Filter sample sheet by index2 length
+    if index2_len is not None:
+        sample_sheet_df = sample_sheet_df.query("index2_len=={}".format(index_len))
+    if sample_sheet_df.shape[0] == 0:
+        logger.error("After filtering for index2 of len '{}' we ended up with no rows in the sample sheet".format(
+            index2_len))
+        sys.exit(1)
+
+    for (sample_type, index_len, index2_len), sample_sheet_type_df in sample_sheet_df.groupby(['Type', 'index_len', 'index2_len']):
         # Modify the sample-sheet
         modified_sample_header_rows, modified_sample_sheet_df = modify_sample_sheet(
             sample_sheet_header_rows=sample_sheet_header_rows,
             sample_sheet_df=sample_sheet_type_df,
             sample_type=sample_type)
 
-        # Set the output file name
-        output_file = sample_sheet_dir / "SampleSheet_{}.csv".format(sample_type)
+        if sample_type is not None and index_len is not None and index2_len is not None:
+            output_file = sample_sheet_dir / "SampleSheet.csv"
```

Comment:
This suggests, the user needs to pass these _three_ arguments to produce `SampleSheet.csv` file - which will be based on specified inputs?

Reply:
These are set by default, if the `-all` parameter is set, these values are set to none and all possible samplesheets grouped by type, index_len, and index2_len are generated.

Reply:
all_arg = getattr(args, 'all')
    if all_arg:
        logger.info("Setting sample-type, index and index2 all to none since the --all parameter was specified")
        setattr(args, "sample_type", None)
        setattr(args, "index_length", None)
        setattr(args, "index2_length", None)

    return args

Reply:
That sounds good!